cloudevents/extensions/recordedtime.md

```diff
@@ -0,0 +1,106 @@
+which is the time when the CloudEvent was created by a producer.
+
+This attribute is distinct from the [`time`
+attribute](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#time),
+which, according to the CloudEvents specification, SHOULD be the time when the
+occurrence happened, if it can be determined.
+
+This attribute makes it possible to represent
+[bitemporal](https://en.wikipedia.org/wiki/Bitemporal_modeling) data with
+CloudEvents so that, for every event, both of the following times can be known
+by consumers:
+
+- _Occurrence time_: timestamp of when the occurrence recorded in the event
+  happened, which corresponds to the `time` attribute.
+- _Recorded time_: the timestamp of when the occurrence was recorded in a
+  specific CloudEvent instance, which is represented by this extension.
+
+## Attributes
+
+### recordedtime
+
+- Type: `Timestamp`
+- Description: Timestamp of when the occurrence was recorded in this CloudEvent,
+  i.e. when the CloudEvent was created by a producer. If this attribute is used
+  the `time` attribute SHOULD be the time when the occurrence happened or the
+  best approximation thereof.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in
+    [RFC 3339](https://tools.ietf.org/html/rfc3339)
+
+## Usage
+
+When this extension is used, producers MUST set the value of the `recordedtime`
+attribute to the timestamp of when they create a new CloudEvent.
+
+If the same occurrence MUST be recorded differently, or the event data or
+attributes of a previous record of the occurrence MUST be amended or redacted,
+then the new CloudEvent with the necessary changes SHOULD have a different
+`recordedtime` attribute value than the previous record of the occurrence.
+
+Intermediaries MUST NOT change the value of the `recordedtime` attribute.
+
+## Use cases
+
+Examples of when an occurrence may need to be recorded differently are:
+
+- When incompatible changes to the event data schema are made, and there are
+  systems that can only process the new schema.
+- When a previous record contains incorrect information.
+- When a previous record contains personal information that may no longer be
+  kept because of regulatory or statutory reasons and should be redacted.
+
+Having bitemporal data makes it easier to get reproducible datasets for
+analytics and data science, as the datasets can be created by placing
+constraints on both the `time` and `recordedtime` attributes of events.
+
+Knowing when an occurrence was recorded in a particular event was recorded also
+makes it possible to determine latency between event producers and consumers. It
+also makes it possible to do operations which are sensitive to the time when an
+event was recorded, such as capturing events into time-intervalled files.
+
+The recorded time also makes it easier to differentiate different records of the
+same occurrence in analytical data stores.
+
+## Terminology
```

Comment:
I'd prefer if we didn't duplicate these terms here because then we need to worry about them getting out of sync.

Reply:
I removed this and just link to the core spec, also happy to remove that, not entirely sure if it is needed, it is somewhat implied already.

Reply:
I don't think this section is needed at all since you already define "recordedtime" above.

Reply:
Removing it, it does not really add much value.